### ISSUE TITLE
Parallax Occlusion Mapping (POM) and Pixel Depth Offset (PDO) Issue

### ISSUE THREAD

**Issue by superkitcath** [o3de/o3de#8349]
[Migrated from JIRA ATOM-15075]
Customer is unable to output a proper shadow map when light is occluded by an object using POM.  One would expect the casted shadow to contain the peaks and valleys of the POM, but instead we only see the flat plane. 

 

Original Forum Post: [https://forums.awsgametech.com/t/10450]
----
Today I had a meeting with Jonny Galloway, Chris Santora and Brodie about Parallax Occlusion Mapping (POM), and Pixel Depth Offset (PDO). We notice some issue with POM/PDO not working properly so we wanted to take this more into investigation. Here's the requested files from Chris Santora.

File Link: [https://drive.google.com/file/d/1puENRt4pPTtszEGzUysCxjUm0bzwUZoS/view?usp=sharing]

We have tried to edit the StandardPBR. Tweaked one of the line in the code: PSDepthOutput

struct PSDepthOutput

{ float m_depth : SV_DepthLessEqual; }

;

to

struct PSDepthOutput

{ float m_depth : SV_DepthGreaterEqual; }

;

but it seems to not make any changes. Here's the image of what it looks like on the bottom, you can see the shadow not being casted from extruded mesh on the bottom plane.

!image-2021-03-26-10-07-55-402.png!

**Comment by smurly** [comment]
this might need an active assignee

**Comment by galibzon** [comment]
@smurly Can you please check if it still occurs. or ask LB to re-validate?

**Comment by galibzon** [comment]
@santorac

**Comment by santorac** [comment]
Yes, I think I'm the developer who is quoted above. As described, the problem is the parallax code does not tolerate a UV tiling rate that isn't 1:1 with world units. Unfortunately I don't have any recommendations beyond what is mentioned above.
